oort/oort-aws/src/main/groovy/com/netflix/spinnaker/oort/aws/provider/agent/ClusterCachingAgent.groovy

```diff
@@ -601,7 +595,10 @@ class ClusterCachingAgent implements CachingAgent, OnDemandAgent, AccountAware,
     )
 
     if (result.autoScalingGroups && !result.autoScalingGroups.isEmpty()) {
-      return result.autoScalingGroups.get(0)
+      def asg = result.autoScalingGroups.get(0)
+
+      // A non-null status indicates that the ASG is in the process of being destroyed
+      return (asg.status == null) ? asg : null
```

Comment:
same question above - not sure what we're doing with the `asg`, but from the comment it seems like we'd want to discard this result, since its status is `null`